sphinx/writers/latex.py

```diff
@@ -65,7 +65,7 @@ class UnsupportedError(SphinxError):
     category = 'Markup is unsupported in LaTeX'
 
 
-class LaTeXWriter(writers.Writer):  # type: ignore[misc]
+class LaTeXWriter(writers.Writer):  # type: ignore[type-arg]
```

Comment:
what is the error that's being suppressed in each of these cases?

Reply:
```
sphinx\writers\latex.py:68:19: error: Missing type parameters for generic type "Writer"  [type-arg]
```

Reply:
i suspect all this needs is

```python
class LaTeXWriter(writers.Writer[str]):
```

Reply:
Right, but *at runtime*, the class isn't generic, so this would yield a TypeError.

Reply:
> Right, but _at runtime_, the class isn't generic, so this would yield a TypeError.

i'm not sure what you mean? no class in Python is generic in this sense at runtime. it's a 'lint time' behaviour only

Reply:
If I did the following to the PR:

```diff
-class LaTeXWriter(writers.Writer):  # type: ignore[type-arg]
+class LaTeXWriter(writers.Writer[str]):
```

then class instantiation etc would fail with a TypeError. Are you suggesting to put something similar in TYPE_CHECKING instead? I don't follow, sorry.

A

Reply:
have you tried it? do you actually get runtime errors?

Reply:
Changing to `Writer[str]` but not removing `# type: ignore[type-arg]`:

```ps1con
PS> mypy
sphinx\io.py:146: error: Unused "type: ignore" comment  [unused-ignore]
sphinx\writers\latex.py:68: error: Unused "type: ignore" comment  [unused-ignore]
sphinx\writers\text.py:359: error: Unused "type: ignore" comment  [unused-ignore]
sphinx\writers\texinfo.py:107: error: Unused "type: ignore" comment  [unused-ignore]
Found 4 errors in 4 files (checked 326 source files)

PS> python -m pytest -vv
ImportError while loading conftest 'S:\Development\sphinx\tests\conftest.py'.
tests\conftest.py:16: in <module>
    from sphinx.testing.util import _clean_up_global_state
sphinx\testing\util.py:17: in <module>
    import sphinx.application
sphinx\application.py:28: in <module>
    from sphinx.registry import SphinxComponentRegistry
sphinx\registry.py:15: in <module>
    from sphinx.io import create_publisher
sphinx\io.py:146: in <module>
    class SphinxDummyWriter(UnfilteredWriter[str]):  # type: ignore[type-arg]
E   TypeError: type 'UnfilteredWriter' is not subscriptable
```

Reply:
i see what you mean...

have asked the typeshed senseis if there's a better way to handle this than just suppressing the error (which seems to obviate any benefit of adding generics to the upstream typestubs)

Reply:
Ok. I'll merge for now & we can revisit later should there be a better approach.

Reply:
There's no pretty solution but here's mypy's general guidance: https://mypy.readthedocs.io/en/stable/runtime_troubles.html#using-classes-that-are-generic-in-stubs-but-not-at-runtime